tracer/test/Datadog.Trace.ClrProfiler.Managed.Tests/AutoInstrumentation/AWS/MemoryStreamCarrierGetter.cs

```diff
@@ -0,0 +1,53 @@
+// <copyright file="MemoryStreamCarrierGetter.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Amazon.SimpleNotificationService.Model;
+using Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.SNSTests;
+using Datadog.Trace.ExtensionMethods;
+using Datadog.Trace.Propagators;
+using Newtonsoft.Json;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.SNSTests
+{
+    public struct MemoryStreamCarrierGetter : ICarrierGetter<PublishRequestProxy>
+    {
+        private const string DatadogKey = "_datadog";
+        private const string CacheKey = "TestCacheKey";  // constant key
+        private static readonly Dictionary<string, Dictionary<string, string>> Cache = new();
```

Comment:
Since this is test code, I advise not using a static cache. It's okay to not cache the results and removing the cache would simplify the test code

Reply:
if I didn't cache it then I would run into a memory stream error, i.e. the stream was closed and we would try to Read it again and it would error out. So I decided to cache the result.

Reply:
I can however look removing the static part

Reply:
When you create your StreamReader, you might be able to use the `bool leaveOpen` [constructor argument](https://learn.microsoft.com/en-us/dotnet/api/system.io.streamreader.-ctor?view=net-7.0#system-io-streamreader-ctor(system-io-stream-system-text-encoding-system-boolean-system-int32-system-boolean)) so you don't dispose the stream the first time